unstable/lens/vector.rkt

```diff
@@ -0,0 +1,50 @@
+#lang racket/base
+
+(provide vector-ref-lens
+         vector-ref-nested-lens
+         vector-pluck-lens
+         )
+
+(require fancy-app
+         lens/base/main
+         "arrow.rkt"
+         "join.rkt"
+         )
+(module+ test
+  (require rackunit))
+
+(define (vector-ref-lens i)
+  (make-lens
+   (vector-ref _ i)
+   (vector-set _ i _)))
```

Comment:
Similar to my comment on the string-ref-lens, I think it would be simpler to build this atop `list-ref-lens` instead of implementing `vector-set`.